TM1py/Services/CellService.py

```diff
@@ -1125,28 +1139,55 @@ def _build_blob_to_cube_process(self, cube_name: str, process_name: str, blob_fi
-        # For Consolidated non-existing-element attempt to write , to trigger error
-        input_statement = f"""
-            If( 
-                ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'N' %
-                ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'AN' %
-                ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'C' % 
-                ElementType('{cube_measure}', '', {variable_cube_measure}) @= '' );
-                    nValue = StringToNumber({value_variable});
-                    {function_str}N(nValue,'{cube_name}',{comma_sep_var_elements});
-            ElseIf( 
+        # For non-existing-element attempt to write to trigger error
+        # For consolidated elements spread if allowed else let fail.
+
+        measure_type_equal = f"ElementType('{cube_measure}', '', {variable_cube_measure}) @= "
+        n_write_types = ["'N'", "'AN'", "'C'", "''"]
+        numeric_write_condition = '% \n'.join([measure_type_equal + possible_type for possible_type in
+                                               n_write_types])
+
+        any_c_element_in_write = '% \n'.join([f"ElementType('{dim}', '', {ele}) @= 'C'" for dim, ele in
+                                              zip(dimensions, dimension_variables)])
+
+        numeric_write_statement_with_spread = f"""
+                nValue = StringToNumber({value_variable});
+                IF({any_c_element_in_write});
+                    CellPutProportionalSpread(nValue,'{cube_name}',{comma_sep_var_elements});
+                ELSE;
+                    {numeric_function_str}(nValue,'{cube_name}',{comma_sep_var_elements});
+                ENDIF;
+                """
+
+        numeric_write_statement_without_spread = f"""
+                nValue = StringToNumber({value_variable});
+                {numeric_function_str}(nValue,'{cube_name}',{comma_sep_var_elements});
+                """
+
+        string_write_condition = f"""
                 ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'S' % 
                 ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'AS' % 
-                ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'AA');
-                    sValue = {value_variable};
-                    CellPutS(sValue,'{cube_name}',{comma_sep_var_elements});
-            EndIf;"""
+                ElementType('{cube_measure}', '', {variable_cube_measure}) @= 'AA'
+                """
+
+        string_write_statement = f"""
+                sValue = {value_variable};
+                CellPutS(sValue,'{cube_name}',{comma_sep_var_elements}); 
+                """
+
+        input_statement = f"""
+        If({numeric_write_condition});
+            {numeric_write_statement_with_spread if allow_spread else numeric_write_statement_without_spread}
+        ElseIf({string_write_condition});
+            {string_write_statement}
+        EndIf;"""
```

Comment:
Would it be reasonable to include an Else to cover the case that neither `numeric_write_condition` nor `string_write_condition` is met?